pages/login/index.js

```diff
@@ -0,0 +1,234 @@
+  const passwordInput = createNode("input", {"type": "password", "id": "password", "style": "margin-top: 10px;", "placeholder": "Password", "class": "inputField", "contenteditable": true, "data-field": "password"})
+
+  const signinbutton = createNode("div", {"class": "button ctaPrimary", "style": "margin-top: 10px", "id": "sign-in-button"}, "Sign in")
+  
+  formDiv.appendChild(emailInput)
+  formDiv.appendChild(passwordInput)
+  formDiv.appendChild(signinbutton)
+
+  signinbutton.addEventListener('click', () => signin())
+}
+
+function add_signup_form() {
+  const formDiv = document.querySelector('#form')
+  formDiv.innerHTML = ''
+
+  const nameInput = createNode("input", {"type": "name", "id": "name", "placeholder": "Name", "class": "inputField", "contenteditable": true, "data-field": "name"})
+  const emailInput = createNode("input", {"type": "email", "id": "email", "style": "margin-top: 10px;", "placeholder": "Email", "class": "inputField", "contenteditable": true, "data-field": "email"})
+  const passwordInput = createNode("input", {"type": "password", "id": "password", "style": "margin-top: 10px;", "placeholder": "Password", "class": "inputField", "contenteditable": true, "data-field": "password"})
+
+
+  const signupbutton = createNode("div", {"class": "button ctaPrimary", "style": "margin-top: 10px", "id": "sign-in-button"}, "Create account")
+  
+  formDiv.appendChild(nameInput)
+  formDiv.appendChild(emailInput)
+  formDiv.appendChild(passwordInput)
+  formDiv.appendChild(signupbutton)
+
+  signupbutton.addEventListener('click', () => signup())
+}
+
+function updateLogin () {
+  //clear current form
+  if (_is_signin_screen) {
+    // update to sign up screen
+    add_signup_form()
+    document.querySelector('#sign-in-type-text').textContent = 'Have an account?'
+    document.querySelector('#sign-in-type-button').textContent = 'Sign in'
+  } else {
+    add_signin_form()
+    document.querySelector('#sign-in-type-text').textContent = 'No account?'
+    document.querySelector('#sign-in-type-button').textContent = 'Sign up'
+  }
+
+  _is_signin_screen = !_is_signin_screen
+}
+
+// start listening for extension specific information
+window.addEventListener("message", receiveMessage)
+function receiveMessage (event) {
+    const message = event.data
+    logInfo(`${_LOG_SCOPE} Receiving message of type ${message['type']}`)
+    switch (message['type']) {
+      case MESSAGE_TYPES.APP_TO_EXTERNAL_SET_EXTENSION_INFO:
+        _extensionId = message['extensionId']
+        _forceServicesHostname = message['forceServicesHostname']
+        break
+      case MESSAGE_TYPES.APP_TO_EXTERNAL_CHECK_IS_LOADED:
+        window.postMessage({'type': MESSAGE_TYPES.EXTERNAL_TO_APP_IS_LOADED})
+        break
+      default:
+        logInfo(`${_LOG_SCOPE} Skipping message of type ${message['type']}`)
+        break
+    }
+}
+window.postMessage({'type': MESSAGE_TYPES.EXTERNAL_TO_APP_IS_LOADED})
```

Comment:
i'm fine with this but just to check i think this stuff is just for us debugging to send the api key to the right extension ,is that correct?